### PR TITLE
3.15.x fix release

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue

https://graviteecommunity.atlassian.net/browse/APIM-XXX

## Description

A small description of what you did in that PR.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-ubklynxpus.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/3-15-x-fix-release/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: release/ci-steps/doc-new-release-notes.mjs [hunk 1]
```diff
@@ -14,25 +14,30 @@ const docApimChangelogFile = `${docApimChangelogFolder}changelog-${versions.trim
...
+
+        return {
+          ...JSON.parse(foundPRCmd.stdout),
+          commitMessage,
+        };
+      } catch (error) {
+        console.error('🚨 Error while searching PR for commit', commitHash, error);
+      }
+    })
+    .reduce((lastP, nextP) => {
```
Comment (ThibaudAV) [inline release/ci-steps/doc-new-release-notes.mjs]: make it sequential

File: release/ci-steps/doc-new-release-notes.mjs [hunk 2]
```diff
@@ -14,25 +14,30 @@ const docApimChangelogFile = `${docApimChangelogFolder}changelog-${versions.trim
...
+  const fetchPrInfo = await gitLogOutput
+    .trim()
+    .split('\n')
+    .map(async (commitLine) => {
+      try {
+        const commitHash = commitLine.substring(0, commitLine.indexOf(' '));
+        const commitMessage = commitLine.substring(commitLine.indexOf(' ') + 1);
+        const foundPRCmd = await $`gh pr list --search "${commitHash}" --state merged --json url,title,number --jq '.[0]'`;
+
+        await new Promise((resolve) => setTimeout(resolve, 100));
```
Comment (gaetanmaisse) [inline release/ci-steps/doc-new-release-notes.mjs]: We need that to not flood GH API?

File: release/ci-steps/doc-new-release-notes.mjs [hunk 3]
```diff
@@ -14,25 +14,30 @@ const docApimChangelogFile = `${docApimChangelogFolder}changelog-${versions.trim
...
+        };
+      } catch (error) {
+        console.error('🚨 Error while searching PR for commit', commitHash, error);
+      }
+    })
+    .reduce((lastP, nextP) => {
+      return lastP.then((lastV) => {
+        return nextP.then((nextV) => [...lastV, nextV]);
+      });
+    }, Promise.resolve([]));
```
Comment (gaetanmaisse) [inline release/ci-steps/doc-new-release-notes.mjs]: With all these sequential promises it looks easier with a simple for loop no?
```suggestion
  const commitLines = gitLogOutput.trim().split('\n');
  const fetchPrInfo = [];

  for (const commitLine of commitLines) {
    const commitHash = commitLine.substring(0, commitLine.indexOf(' '));
    const commitMessage = commitLine.substring(commitLine.indexOf(' ') + 1);

    try {
      const foundPRCmd = await $`gh pr list --search "${commitHash}" --state merged --json url,title,number --jq '.[0]'`;

      fetchPrInfo.push({
        ...JSON.parse(foundPRCmd.stdout),
        commitMessage,
      });
    } catch (error) {
      console.error('🚨 Error while searching PR for commit', commitHash, error);
    }

    await new Promise((resolve) => setTimeout(resolve, 100));
  }
```

File: release/ci-steps/doc-new-release-notes.mjs (hunk 2 above)
Comment (ThibaudAV) [reply release/ci-steps/doc-new-release-notes.mjs]: yes

File: release/ci-steps/doc-new-release-notes.mjs (hunk 3 above)
Comment (ThibaudAV) [reply release/ci-steps/doc-new-release-notes.mjs]: haha yes 🚀

**Pull request merged**